stem/model.py

```diff
@@ -74,33 +75,245 @@ def all_model_parts(self) -> List[ModelPart]:
+        """
+        Computes the vertical offset based on sleeper parameters.
+
+        If sleeper_parameters is a SoilMaterial and dimensions are provided,
+        returns the sleeper height. Otherwise, returns 0.
+
+        Args:
+        - sleeper_parameters (Union[NodalConcentrated, SoilMaterial]): Sleeper parameters.
+        - sleeper_dimensions (Optional[Sequence[float]]): Sleeper dimensions [length, width, height].
+
+        Returns:
+            float: Vertical offset.
+        """
+        if isinstance(sleeper_parameters, SoilMaterial):
+            if sleeper_dimensions is None:
+                raise ValueError("If sleeper parameters are SoilMaterial, dimensions must be a list of length, width, "
+                                 "height.")
+            return sleeper_dimensions[2]
+        return 0.0
+
+    def _generate_sleepers(self, sleeper_parameters: Union[NodalConcentrated, SoilMaterial],
+                           sleeper_dimensions: Sequence[float], base_sleeper_name: str,
+                           sleeper_global_coords: ndarray[Any, Any], sleeper_rail_pad_offset: float) -> None:
+        """
+        Generates sleeper geometry based on the type of sleeper parameters.
+
+        For NodalConcentrated sleepers, creates point-based geometries.
+        For SoilMaterial sleepers, creates 3D volumes.
+
+        Args:
+            sleeper_parameters (Union[NodalConcentrated, SoilMaterial]): Sleeper parameters.
+            sleeper_dimensions (Sequence[float]): Dimensions for the sleeper if applicable.
+            base_sleeper_name (str): Base name for sleepers.
+            sleeper_global_coords (np.ndarray): Global coordinates for sleeper placement.
+            sleeper_rail_pad_offset (float): Offset between the sleeper end and the rail pad location.
+
+        Returns:
+            None
+
+        """
+        if isinstance(sleeper_parameters, NodalConcentrated):
+            connection_geo_settings = {"": {"coordinates": sleeper_global_coords, "ndim": 1}}
+            self.gmsh_io.generate_geometry(connection_geo_settings, "")
+            # For nodal sleepers, create a connection line and a point geometry for the sleeper.
+            sleeper_geo_settings = {base_sleeper_name: {"coordinates": sleeper_global_coords, "ndim": 0}}
+            self.gmsh_io.generate_geometry(sleeper_geo_settings, "")
+        elif isinstance(sleeper_parameters, SoilMaterial):
+            # if no soil is present then this can be skipped
+            if len(self.body_model_parts) > 0:
+                # get the min and max coordinates of the soils
+                min_coords, max_coords = self.get_bounding_box_soil()
+                # vertical axis should be the same as the initial sleeper height
+                level_vertical_sleeper = sleeper_global_coords[0][VERTICAL_AXIS]
+                level_x_axis = sleeper_global_coords[0][0]
+                min_coords[0] = level_x_axis
```

Comment:
please do not change the direction of the track

Reply:
corrected by adding the direction of the track

Reply:
also added test for the x direction